src/sp/it/pl/util/file/Util.kt

```diff
@@ -72,24 +72,36 @@ fun File.isAnyChildOf(parent: File) = parent.isAnyParentOf(this)
  * @return child files of the directory or empty if parameter null, not a directory or I/O error occurs
  */
 @Suppress("DEPRECATION")
-fun File.listChildren(): Stream<File> = listFiles()?.asSequence()?.asStream() ?: Stream.empty()
+fun File.listChildren(): Stream<File> = 
+        listFiles().asStream()
 
 @Suppress("DEPRECATION")
-fun File.seqChildren(): Sequence<File> = listFiles()?.asSequence() ?: sequenceOf()
+fun File.listChildren(filter: FileFilter): Stream<File> = 
+        listFiles(filter).asStream()
 
 @Suppress("DEPRECATION")
-fun File.listChildren(filter: FileFilter): Stream<File> = listFiles(filter)?.asSequence()?.asStream() ?: Stream.empty()
+fun File.listChildren(filter: FilenameFilter): Stream<File> =
+        listFiles(filter).asStream()
 
 @Suppress("DEPRECATION")
-fun File.listChildren(filter: FilenameFilter): Stream<File> = listFiles(filter)?.asSequence()?.asStream() ?: Stream.empty()
+fun File.seqChildren(): Sequence<File> =
+        listFiles()?.asSequence().orEmpty()
+
+/** @return a Stream built from this array, which is empty if the array is null */
+fun <T> Array<T>?.asStream() = this?.asSequence()?.asStream() ?: Stream.empty()
+
+fun <T> Stream<T>?.orEmpty() = this ?: Stream.empty()
+
+fun <T> Sequence<T>?.orEmpty() = this ?: emptySequence()
 
 /**
  * Safe version of [File.getParentFile].
  *
  * @return parent file or null if is root
  */
 @Suppress("DEPRECATION")
-val File.parentDir: File? get() = parentFile
+val File.parentDir: File?
```

Comment:
I do not really understand the formatting of the get() being on the new line. Is this some code style you apply or your preferred way of writing getters?

Reply:
It is the way auto formatting does it, using the shared code style in the project. But in my opinion it's more readable that way anyways.

Reply:
You know you can do this with one click, right? But don't do it while we have active branches, it would introduce a lot of unnecessary conflicts